### PR TITLE
Allow setting visionOS deployment target for appleVisionWithiPadDesign

### PR DESCRIPTION
**Pull request by freak4pc**
### Short description 📝

I reported this issue on Slack and discussed with @waltflanagan, when setting `appleVisionWithiPadDesign` - you should be able to set a `visionOS`-specific deployment target.

Otherwise, Xcode defaults to the latest one, which could be 1.1 (even if you want 1.0, for example).

### How to test the changes locally 🧐

- Create a project that has iPhone, iPad and appleVisionWithiPadDesign Destinations. 
- Set a multiplatform deployment target for iOS & visionOS.
- Before this fix, this yields this error:

image

I've added tests to cover these areas.

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistGraph/Models/Destination.swift
Comment (waltflanagan) [inline Sources/TuistGraph/Models/Destination.swift:42]: Changing the association with `platform` will cause issues that diverge with what Xcode derives from these settings.  If a target has a single destination of `appleVisionWithiPadDesign` the `SDKROOT` is `iphoneos` but this would cause tuist to use `xros` instead.   Additionally, `SUPPORTED_PLATFORMS` would resolve to `xros xrsimulator`.

There are a few other values that are derived from this value, but making a this change would be a meaningful change to how `tuist` generates targets and I dont think this is the right approach to allowing `XROS_DEPLOYMENT_TARGET` to be set for `appleVisionWithiPadDesign ` targets.

I would expect that adjusting the [TargetLinter](https://github.com/tuist/tuist/blob/362ef5e3c35fda201960ecadc5edde9ef0ca5734/Sources/TuistGenerator/Linter/TargetLinter.swift#L208) to account for this need is a more narrow change to accomplish the goal.

**Review by waltflanagan — CHANGES_REQUESTED** [review]

File: Sources/TuistGraph/Models/Destination.swift
Comment (freak4pc) [reply Sources/TuistGraph/Models/Destination.swift:42]: I ran some tests and it worked absolutely fine, but generally I agree - I wanted to do the more narrow change in TargetLinter but because it checks platform.supports, it will mean I manually have to check against the target's destinations which also seems a bit gnarly.

I'll add another commit with that option.

**Comment by freak4pc** [comment]
@waltflanagan Check it out when you have a moment. Thanks!

**Comment by pepicrft** [comment]
> Uhm, the tests are passing locally and I can't enter Codemagic to see what it failed on. Isn't it an OSS-centric CI that everyone can see the build failures?
> Any thoughts?

I'm looking into the Codemagic authorization issue. Regarding the failing builds, it's also a known issue with the Tuist Cloud API. We are looking into it. To prevent it from blocking the merge of this PR, I'll run the workflows locally before merging once @waltflanagan confirms the changes are fine.

File: Tests/TuistGeneratorTests/Generator/ConfigGeneratorTests.swift
Comment (waltflanagan) [inline Tests/TuistGeneratorTests/Generator/ConfigGeneratorTests.swift:545]: It would be nice to also include a check for `"SDKROOT": "iphoneos",` as well

File: Tests/TuistGeneratorTests/Linter/TargetLinterTests.swift
Comment (waltflanagan) [inline Tests/TuistGeneratorTests/Linter/TargetLinterTests.swift:277]: The name here does not match what we're checking for.  I would expect `is_invalid` to check that we got linting errors, but we're using `XCTDoesNotContainLintingIssue` to ensure we did not get them

File: Sources/TuistGenerator/Linter/TargetLinter.swift
Comment (waltflanagan) [inline Sources/TuistGenerator/Linter/TargetLinter.swift:226]: 👍 this is about what i was expecting.

File: Tests/TuistGeneratorTests/Linter/TargetLinterTests.swift
Comment (freak4pc) [reply Tests/TuistGeneratorTests/Linter/TargetLinterTests.swift:277]: woopsie :)

Comment (freak4pc) [reply Tests/TuistGeneratorTests/Linter/TargetLinterTests.swift]: lmk your thoughts

**Review by waltflanagan — APPROVED** [review]

**Review by danieleformichelli — APPROVED** [review]

**Pull request merged**
